### PR TITLE
feat(sdk): add `Bucket.copy()` for `tf-gcp` platform

### LINKED ISSUE #4398: Implement `cloud.Bucket` inflight method `copy` for `tf-gcp` target
```js
/**
 * Copy an object to a new location in the bucket. If the destination object
 * already exists, it will be overwritten.
 *
 * @throws Will throw if the `src` object doesn't exist.
 */
inflight copy(src: str, dst: str): void;
```

### PR DESCRIPTION
**Pull request by garysassano**
Closes #4398

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by garysassano** [comment]
Tested to work on GCP.

**Comment by garysassano** [comment]
Keep.

File: libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts [hunk 1]
```diff
@@ -310,5 +310,27 @@ test("tryDelete a non-existent object from the bucket", async () => {
...
+
+//   // WHEN
+//   const client = new BucketClient(BUCKET_NAME, mockStorage as any);
+//   const response1 = await client.copy(SRC_KEY, SRC_KEY);
+//   const response2 = await client.copy(SRC_KEY, DST_KEY);
+
+//   // THEN
+//   expect(response1).toEqual(undefined);
+//   expect(response2).toEqual(undefined);
+// });
```
Comment (tsuf239) [inline libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts:333]: Do you need help in overcoming this? 
what about spying on the `client.bucket.file.copy` method and check if the inner gcp command is called with the right parameters, then mocking the resolved value correctly?

File: libs/wingsdk/src/shared-gcp/bucket.inflight.ts [hunk 2]
```diff
@@ -27,9 +27,14 @@ export class BucketClient implements IBucketClient {
...
-      `copy is not implemented: (srcKey=${srcKey}, dstKey=${dstKey})`
-    );
+    try {
+      const srcFile = this.bucket.file(srcKey);
+      await srcFile.copy(this.bucket.file(dstKey));
+    } catch (error) {
+      throw new Error(
+        `Failed to copy object from (srcKey=${srcKey}) to (dstKey=${dstKey}).`
+      );
+    }
```
Comment (tsuf239) [inline libs/wingsdk/src/shared-gcp/bucket.inflight.ts:37]: can you add it to `_supportedOps` too?

File: libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts (hunk 1 above)
Comment (garysassano) [reply libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts:391]: How do you spy on a method that doesn't exist?

Comment (tsuf239) [reply libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts:391]: does `client.bucket.file` exist? 
if so, I guess you can do `client.bucket.file.copy = vi.fn`
then spying on it :)

**Comment by garysassano** [comment]
Test passes on GCP, even tho the `copy()` method of [mock-gcs](https://www.npmjs.com/package/mock-gcs) module doesn't work.

image

**Comment by garysassano** [comment]
Test passes on GCP, even tho the `copy()` method of [mock-gcs](https://www.npmjs.com/package/mock-gcs) module doesn't work.

image

**Review by tsuf239 — APPROVED** [review]

**Pull request merged**